kats/src/main/kotlin/kats/Option.kt

```diff
@@ -24,6 +24,11 @@ package kats
  */
 sealed class Option<out A> {
 
+    companion object {
+
+        inline operator fun <A> invoke(f: () -> A): Option<A> = f().let { if (it == null) None else Some(it) }
```

Comment:
This should not be a function but a strict non nullable argument. This factory method would violate the laws because if you pass in `null` you should get `Some(null)`. You are adding implicitly here a safeguard that may not be wanted by the user.